contracts/its-borsh-translator/src/borsh.rs

```diff
@@ -0,0 +1,863 @@
+use axelar_wasm_std::{nonempty, IntoContractError};
+use borsh::{BorshDeserialize, BorshSerialize};
+use cosmwasm_std::{HexBinary, Uint256};
+use error_stack::{Report, ResultExt};
+use interchain_token_service_std::TokenId;
+use router_api::ChainNameRaw;
+
+#[derive(thiserror::Error, Debug, IntoContractError)]
+pub enum Error {
+    #[error("borsh serialization failed")]
+    SerializationFailed,
+    #[error("borsh deserialization failed")]
+    DeserializationFailed,
+    #[error("invalid chain name")]
+    InvalidChainName,
+    #[error(transparent)]
+    NonEmpty(#[from] nonempty::Error),
+}
+
+/// Borsh-serializable mirror of interchain_token_service_std::InterchainTransfer
+#[derive(BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq)]
+pub struct InterchainTransfer {
+    pub token_id: [u8; 32],
+    pub source_address: Vec<u8>,
+    pub destination_address: Vec<u8>,
+    pub amount: [u8; 32], // Uint256 as 32-byte little-endian
```

Comment:
Could we also parse and transmit this as `u64`? That would save 24 bytes.

Reply:
got it. yes Solana SPL tokens have a maximum amount of `u64::max`. however it's true this encoding should work for other chains as well.